src/briefcase/integrations/android_sdk.py

```diff
@@ -212,61 +248,305 @@ def verify_emulator(self):
+            print()
+            print("    briefcase run android -d {device}".format(device=device))
+
+        return device, name, avd
+
+    def create_emulator(self):
+        """Create a new Android emulator.
+
+        """
+        print()
+        name = self.command.input("Emulator name: ")
+
+        raise BriefcaseCommandError(
+            """
+You can create an emulator by running:
 
     $ {sdkmanager_path} "platforms;android-28" \
 "system-images;android-28;default;x86" "emulator" "platform-tools"
 
     $ {avdmanager_path} --verbose create avd \
---name robotfriend --abi x86 \
+--name {name} --abi x86 \
 --package 'system-images;android-28;default;x86' --device pixel
 
-    $ echo 'disk.dataPartition.size=4096M' >> $HOME/.android/avd/robotfriend.avd/config.ini
-
-    $ {emulator_path} -avd robotfriend &
+    $ echo 'disk.dataPartition.size=4096M' >> $HOME/.android/avd/{name}.avd/config.ini
 
 """.format(
-            adb_path=adb_path,
-            avdmanager_path=avdmanager_path,
-            emulator_path=emulator_path,
-            sdkmanager_path=sdkmanager_path,
+                sdkmanager_path=self.sdkmanager_path,
+                avdmanager_path=self.avdmanager_path,
+                name=name,
+            )
         )
 
+    def start_emulator(self, avd):
+        """Start an existing Android emulator.
+
+        Returns when the emulator is booted and ready to accept apps.
+
+        :param avd: The AVD of the device.
+        """
+        if avd in set(self.emulators()):
+            raise BriefcaseCommandError(
+                """
+You can start the emulator by running:
+
+    $ {emulator_path} -avd {avd} &
+
+""".format(
+                    emulator_path=self.emulator_path, avd=avd
+                )
+            )
+        else:
+            raise InvalidDeviceError("emulator AVD", avd)
+
 
 class ADB:
     def __init__(self, android_sdk, device):
         """
         An API integration for the Android Debug Bridge (ADB).
 
         :param android_sdk: The Android SDK providing ADB.
-        :param device: The ID of the device to target (in a format usable by`adb -s`)
+        :param device: The ID of the device to target (in a format usable by
+            `adb -s`)
         """
         self.android_sdk = android_sdk
         self.command = android_sdk.command
         self.device = device
 
+    def avd_name(self):
+        """Get the AVD name for the device.
+
+        :returns: The AVD name for the device; or ``None`` if the device isn't
+            an emulator
+        """
+        try:
+            output = self.run("emu", "avd", "name")
+            return output.split("\n")[0]
+        except subprocess.CalledProcessError as e:
+            # Status code 1 is a normal "it's not an emulator" error response
+            if e.returncode == 1:
+                return None
+            else:
+                raise BriefcaseCommandError(
```

Comment:
Because we call this from other helper functions, in a way that the user isn't really aware of, I'd prefer that we not crash the `briefcase` process in this case. Not a showstopper.

Reply:
A reasonable concern; but in this case, if you ask for an AVD of a non-existent device ID, something has gone *badly* wrong.